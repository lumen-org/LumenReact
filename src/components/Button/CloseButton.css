
.close-container{
  position: relative;
  /* margin: auto; */
  width: 32px;
  height: 24px;
  margin: 0;
  padding: 0.1rem 0.5rem 0.1rem 0.5rem;
  cursor: pointer;
}

.leftright{
  height: 0.188rem;
  width: 1.172rem;
  position: absolute;
  margin-top: 9.8px;
  background-color: #bbbbd4;
  border-radius: 1px;
  transform: rotate(45deg);
  transition: all .4s ease-in;
}

.rightleft{
  height: 0.188rem;
  width: 1.172rem;
  position: absolute;
  margin-top: 9.8px;
  background-color: #bbbbd4;
  border-radius: 1px;
  transform: rotate(-45deg);
  transition: all .4s ease-in;
}


.close-container:hover .leftright{
  transform: rotate(-45deg);
  background-color: #F25C66;
}
.close-container:hover .rightleft{
  transform: rotate(45deg);
  background-color: #F25C66;
}
.close-container:hover label{
  opacity: 1;
}