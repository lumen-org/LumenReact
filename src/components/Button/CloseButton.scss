@import "src/assets/styles/colors";

.close-container{
  position: relative;
  /* margin: auto; */
  width: 32px;
  height: 24px;
  margin: 0;
  padding: 0.1rem 0.5rem 0.1rem 0.5rem;
  cursor: pointer;
}

.leftright{
  height: 0.188rem;
  width: 1.172rem;
  position: absolute;
  margin-top: 9.8px;
  background-color: $close-button-background-color;
  border-radius: 1px;
  transform: rotate(45deg);
  transition: all .4s ease-in;
}

.rightleft{
  height: 0.188rem;
  width: 1.172rem;
  position: absolute;
  margin-top: 9.8px;
  background-color: $close-button-background-color;
  border-radius: 1px;
  transform: rotate(-45deg);
  transition: all .4s ease-in;
}


.close-container:hover .leftright{
  transform: rotate(-45deg);
  background-color: $close-button-container-background-color;
}
.close-container:hover .rightleft{
  transform: rotate(45deg);
  background-color: $close-button-container-background-color;
}
.close-container:hover label{
  opacity: 1;
}