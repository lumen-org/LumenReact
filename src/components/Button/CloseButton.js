import React from "react";
import "./CloseButton.scss";

function CloseButton({ handleClose }) {
  return <div className="close-container" onClick={handleClose}>
  <div className="leftright"></div>
  <div className="rightleft"></div>
  </div>


}

export default CloseButton
