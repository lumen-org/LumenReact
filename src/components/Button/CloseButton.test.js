import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import TestRenderer from "react-test-renderer"
import CloseButton from "./CloseButton";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("CloseButton Missing Values", () => {
  test("CloseButton missing value", () => {
    jest.spyOn(console, "error")
    act(() => {
      render(<CloseButton />, container);
    });
    expect(console.error).toBeCalled()
  }); 
});

describe("CloseButton Functionality", () => {
  test("CloseButton render normally", () => {  
    
    let handleClose = jest.fn();
    let component = TestRenderer.create(<CloseButton handleClose={handleClose} />, container);
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

