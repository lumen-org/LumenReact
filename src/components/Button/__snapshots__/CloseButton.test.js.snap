// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton render normally 1`] = `
<div
  className="close-container"
  onClick={[MockFunction]}
>
  <div
    className="leftright"
  />
  <div
    className="rightleft"
  />
</div>
`;
